### PR TITLE
Move regression tests that use C library functions to a separate folder

### PR DESCRIPTION
**Pull request by tautschnig**
This is a first step to disentangle testing of CBMC's algorithms and core
components from testing of the library models. The next step is to ensure
coverage of all modelled library functions.

Marking this do-not-merge as there are further tests to be moved, and some tests to be split into multiple, smaller tests. Nevertheless, early feedback as to whether this is a good idea at all would be much appreciated.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by martin-cs** [comment]
> Nevertheless, early feedback as to whether this is a good idea at all would be much appreciated.

[+] This fits better with the development model that seems to be happening where library models are increasingly independent and their own repositories.
[+] Clarity about what is being tested is always a good thing.
[-] How intrusive are this patch set likely to be?

**Comment by tautschnig** [comment]
I'd now consider this ready for review. In a subsequent step I'll propose new regression tests for library functions currently not being exercised.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 254d09782f3c893ecd6044e2d2709a8fc28d00df).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/89812749

**Comment by martin-cs** [comment]
Views mostly as above but a PR this size should really get buy-in from multiple people.  @chrisr-diffblue @smowton @peterschrammel what do you think?

**Comment by smowton** [comment]
The idea seems sensible to me, but I can't see myself having time to provide a meaningful review

File: regression/cbmc-library/CreateThread1/main.c [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include <windows.h>
```
Comment (chrisr-diffblue) [inline regression/cbmc-library/CreateThread1/main.c]: Minor nit pick - could we slightly tweak the directory naming for these test? having a number postfix with no obvious separator gets a bit confusing where the function being tested *also* has a numeric postfix (e.g. `_InterlockedAnd161` is the first test of  the `_InterlockedAnd16` function) - perhaps a naming scheme like `functionUnderTest-N` might be clearer? Or maybe even sub-directories, but thats probably more mechanics than we'd want to get into.

**Comment by chrisr-diffblue** [comment]
I'm really pleased to see some cleanup to our tests, which are quite a mess to navigate - so I'm definitely supportive. One minor nit pick, but thats all. (Thanks @martin-cs for flagging up this PR)

**Review by chrisr-diffblue — APPROVED** [review]

File: regression/cbmc-library/CreateThread1/main.c (hunk 1 above)
Comment (tautschnig) [reply regression/cbmc-library/CreateThread1/main.c]: Indeed, that was very confusing. I have now used a `-01` suffix instead of `1`.

**Comment by tautschnig** [comment]
As reviewing the last commit in this PR is obviously somewhat pointless, here is the script that I've used to generate the tests:
````bash
#!/bin/bash

for lib in src/ansi-c/library/*.c src/cpp/library/*.c
do
  echo "lib: $lib"
  header=$(basename $lib .c)
  echo "header: $header"

  for f in $(git grep FUNCTION: $lib | cut -f3 -d" " | sort | uniq)
  do
    echo "f: $f"
    test_dir="regression/cbmc-library/${f}-01"
    if [ ! -d $test_dir ]
    then
      mkdir -p $test_dir

      cat > $test_dir/main.c <<EOF
#include <$header.h>
#include <assert.h>

int main()
{
  $f();
  assert(0);
  return 0;
}
EOF
      git add $test_dir/main.c

      cat > $test_dir/test.desc <<EOF
KNOWNBUG
main.c
--pointer-check --bounds-check
^EXIT=0$
^SIGNAL=0$
^VERIFICATION SUCCESSFUL$
--
^warning: ignoring
EOF
      git add $test_dir/test.desc
    fi
  done
done
````

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 6d852fb62145639c8d3bf24a2394f7ce2020369a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90469383
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 30ce180f14d49eb54f4c3abca9a0d3897da94ab2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90470682
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: e7330a04f9d195672e702d89c264106aaedff017).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90472111

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: f37b54bfbb778061f43bc843dfdb70b08387a984).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90492359

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 5ec635030011c627bd3756082f482b2f7b07892d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90496872
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: e8b00cd18ec1cb2bbf8b839798002ea391014bd9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90498444

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2b0ad13e6cd9c5683d8e365b1517da45111c5fb5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90525242

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 3534b823b8fc74c98eb5c85233211728d45c05b2).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90583796
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 122c57cd1cd120082d6780b63533c48b9209d597).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90586487
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 61062cb8ed7d187b03c86833b5303dbc2e07d39a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90588740
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: dfea0a69a3bd69fb38841fffeecec6c377c46fab).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90587440
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: a4d2828ffafe17801c9807ca301404ee399bcf4f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/90854570

**Pull request merged**